### DISCUSSION TITLE
Pre-install CLI dependencies in Windmill workers?

### DISCUSSION THREAD

**Discussion by pcgeek86** [windmill-labs/windmill#1812] [Questions] [answered]
I need certain command line utilities available in my Windmill scripts. However, I have to install them every time the script runs. Is there a way to pre-install utilities, so that I don't have to install them on every execution?

Is it possible to build and use a custom Docker image for the workers?

**Comment by SindreSvendby** [comment]
Yes. Custom images is possible. 

See https://discord.com/channels/930051556043276338/930051556043276341/1127862863969652746

And 
https://github.com/zsnmwy/windmill-advanced-image

I suggest to log into discord and search. It has been up there a couple of times.

**Comment by rubenfiszel** [comment, marked as answer]
As @SindreSvendby said, you can build a custom image using the windmill image as base (e.g: https://github.com/windmill-labs/windmill/blob/v1.127.0/docker/DockerfileHelm).

We need however to improve the docs in this respect so thank you for bringing it to our attention
